js/src/relay/WalletLinkRelay.ts

```
@@ -840,4 +848,29 @@ export class WalletLinkRelay implements WalletLinkRelayAbstract {
       }
     })
   }
+
+  public async makeEthereumJSONRPCRequest(
+    request: JSONRPCRequest,
+    jsonRpcUrl: string
```

This still requires that the jsonRpcUrl is set.  I think its preferable to make the jsonRpcUrl optional.  Let's make this call fail if it's not provided.

---

Good callout. I left throwing the error inside the provider. It makes more sense to have it here.

---

(3301:5214:205ce2997062)

---

Sorry I mean can we make jsonRpcUrl optional.

Like
```
public async makeEthereumJSONRPCRequest(
    request: JSONRPCRequest,
    jsonRpcUrl?: string
)
```

---

Also can we add a return type for this?

---

I just pushed the change for that!

---

of course! let me add the return type